osu.Game/Graphics/UserInterface/OsuMenu.cs

```diff
@@ -42,6 +43,16 @@ private void load(AudioManager audio)
             sampleClose = audio.Samples.Get(@"UI/dropdown-close");
         }
 
+        protected override void Update()
+        {
+            base.Update();
+
+            bool showCheckboxes = Items.Any(i => i is StatefulMenuItem);
```

Comment:
The unfortunate truth is that this allocates out the rear:

![JetBrains Rider-EAP 2024-10-04 at 08 50 38](https://github.com/user-attachments/assets/42cc568c-4f32-4ec5-ad32-9d4f5c526829)

I'm fine with a non-LINQ version though:

```diff
diff --git a/osu.Game/Graphics/UserInterface/OsuMenu.cs b/osu.Game/Graphics/UserInterface/OsuMenu.cs
index 719100e138..6e7dad2b5f 100644
--- a/osu.Game/Graphics/UserInterface/OsuMenu.cs
+++ b/osu.Game/Graphics/UserInterface/OsuMenu.cs
@@ -3,7 +3,6 @@
 
 #nullable disable
 
-using System.Linq;
 using osu.Framework.Allocation;
 using osu.Framework.Audio;
 using osu.Framework.Audio.Sample;
@@ -47,10 +46,19 @@ protected override void Update()
         {
             base.Update();
 
-            bool showCheckboxes = Items.Any(i => i is StatefulMenuItem);
+            bool showCheckboxes = false;
 
-            foreach (var drawableItem in ItemsContainer.OfType<DrawableOsuMenuItem>())
-                drawableItem.ShowCheckbox.Value = showCheckboxes;
+            foreach (var drawableItem in ItemsContainer)
+            {
+                if (drawableItem.Item is StatefulMenuItem)
+                    showCheckboxes = true;
+            }
+
+            foreach (var drawableItem in ItemsContainer)
+            {
+                if (drawableItem is DrawableOsuMenuItem osuItem)
+                    osuItem.ShowCheckbox.Value = showCheckboxes;
+            }
         }
 
         protected override void AnimateOpen()

```